### PR TITLE
Go 1.7

### PR DESCRIPTION
**Pull request by samuelkarp**
### Summary
Moves us to Go 1.7 and fixes broken tests
Fixes https://github.com/aws/amazon-ecs-agent/issues/485

### Implementation details
Go 1.7 started failing tests where a goroutine failed after the test ended.  Some of our tests were doing this; they've now been fixed.

### Testing
- [x] Builds (`make release`)
- [x] Unit tests (`make short-test`) pass
- [x] Integration tests (`make test` and `make run-integ-tests`) pass
- [ ] Functional tests (`make run-functional-tests`) pass

New tests cover the changes: no, but old ones are fixed

### Description for the changelog
No changelog impact

### Licensing
This contribution is under the terms of the Apache 2.0 License: yes (Amazon employee)

### CODE REVIEW

**Comment by samuelkarp** [comment]
`TestSendsEventsTaskDifferences` looks to be timing out when running on an EC2 instance.  I'll look at that tomorrow and update the pull request.

**Comment by samuelkarp** [comment]
I've fixed the issue with `TestSendsEventsTaskDifferences`.

r? @aaithal @richardpen @nmeyerhans

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: go
 sudo: false
 go:
-    - 1.6
+    - 1.7
```
Comment (aaithal) [inline .travis.yml]: It seems like copyright headers need to be updated for all of the files in this commit with the current year.

Comment (samuelkarp) [inline .travis.yml]: Done

**Pull request merged**